File: baekjoon/10040.cpp
#include <bits/stdc++.h>
#define loop(i,n) for(int i=0;i<n;++i)
using namespace std;

int cost[1000], vote[1000];
int main()
{
    int N, M; cin >> N >> M;
    loop(i, N) cin >> cost[i];
    while (M--) {
        int need; cin >> need;
        loop(i,N)
            if (cost[i] <= need) {
                ++vote[i];
                break;
            }
    }
    
    int max_vote = 0, ans;
    loop(i, N)
        if (vote[i] > max_vote) {
            max_vote = vote[i];
            ans = i + 1;
        }
    cout << ans;
    return 0;
}

File: baekjoon/10000/10040.rs
use std::io::*;

struct Scanner {
    it: std::str::SplitAsciiWhitespace<'static>,
}

impl Scanner {
    fn new() -> Self {
        let mut s = String::new();
        stdin().read_to_string(&mut s).ok();
        Self { it: s.leak().split_ascii_whitespace() }
    }
    fn read<T: std::str::FromStr>(&mut self) -> T {
        self.it.next().unwrap().parse::<T>().ok().unwrap()
    }
    fn read_vec<T: std::str::FromStr>(&mut self, len: usize) -> Vec<T> {
        (0..len).map(|_| self.read()).collect()
    }
}

fn main() {
    let mut sc = Scanner::new();
    let mut bw = BufWriter::new(stdout().lock());
    let n = sc.read::<usize>();
    let m = sc.read::<usize>();
    let cost = sc.read_vec::<i32>(n);
    let mut vote = vec![0; n];
    for _ in 0..m {
        let need = sc.read::<i32>();
        let i = cost.iter().position(|&c| c <= need).unwrap();
        vote[i] += 1;
    }
    let (ans, _) = vote.into_iter().enumerate().max_by(|l, r| l.1.cmp(&r.1)).unwrap();
    writeln!(bw, "{}", ans + 1).ok();
}
